### DISCUSSION TITLE
404? NOT serving HTTPS

### DISCUSSION THREAD

**Discussion by Senthie** [coder/code-server#4884] [Q&A]
您好，当我在树莓派4B上运行code-server，并在浏览器上输入地址的时候返回的是404，我认为是NOT serving HTTPS的问题，但是我并不是该领域的研究人员，大家有什么解决办法吗
![q1](https://user-images.githubusercontent.com/78528866/154739203-86afe50c-8ffd-48cc-9965-0e214abd0faf.jpg)
![q2](https://user-images.githubusercontent.com/78528866/154739314-64376b75-4fb9-477b-9531-71336b113972.png)

**Comment by jsjoeio** [comment]
Hmm...I'm not sure why it's 404ing on your Pi. Is it installed with yarn/npm? 

Are there any logs in the browser console? (Inspect > Console)

**Comment by cxw620** [comment]
> 您好，当我在树莓派4B上运行code-server，并在浏览器上输入地址的时候返回的是404，我认为是NOT serving HTTPS的问题，但是我并不是该领域的研究人员，大家有什么解决办法吗 ![q1](https://user-images.githubusercontent.com/78528866/154739203-86afe50c-8ffd-48cc-9965-0e214abd0faf.jpg) ![q2](https://user-images.githubusercontent.com/78528866/154739314-64376b75-4fb9-477b-9531-71336b113972.png)

这个好像是你不指定code-server的证书不开启https就显示这个NOT serving HTTPS. 访问一下`127.0.0.1:8189`试一下?

**Reply by jsjoeio** [reply]
Can you translate your message to English? Then other folks might be able to help as well

**Reply by WeAreAllLonlyGhost** [reply]
same problem.
use code-server -vvv to start it ,then access code-server,you will see some logs.

**Reply by jsjoeio** [reply]
@WeAreAllLonlyGhost can you post your logs? I'm using 4.0.2 without issue

**Reply by Senthie** [reply]
I tried using 127.0.0.1:8189, but still 404,I don't know if code-server -vvvv works right. I didn't find any useful information in the browser log file (maybe I don't know how to use it), code-server is installed with yam, I remember there is a version of arml7, but it doesn't work on rpi4B

**Reply by jsjoeio** [reply]
Can you try installing the beta version with `yarn global add code-server@beta` and trying that?

**Comment by VincentWTS** [comment]
I have encountered same issue, and after I use the beta version and use code-server -vvv to start it, I see the log below:

It said "Extension VSIX downlads cache dir does not exist".

<img width="721" alt="image" src="https://user-images.githubusercontent.com/78941927/156687508-4fac09b3-e3a8-4e75-b7a6-0065ca6bfdcc.png">

**Reply by jsjoeio** [reply]
Can you try upgrading to 4.1.0 to see if the issue is resolved?

**Reply by VincentWTS** [reply]
It is resolved in 4.1.0. Thanks a lot!!

**Reply by jsjoeio** [reply]
Hooray!!!! Thanks for confirming. Happy coding!
